### ISSUE TITLE
[ENH] Inputs w/require attribute -> raise warning on set event, iff require not yet met.

### ISSUE THREAD

**Issue by oesteban** [nipy/nipype#1143]
Currently, if an interface is imported and it contains inputs marked with requires=['field'], a warning is always raised. For instance:

```
nipype/nipype/interfaces/base.py:403: UserWarning: Input icorr requires inputs: shift_in_file warn(msg)
```

**Comment by satra** [comment]
i feel now that user warnings are mostly useless. we could remove the warning and just have the check_inputs function raise errors in such situations.

**Comment by oesteban** [comment]
Definitely. Do you think it is worth spending time on checking when setting an input with requires?

**Comment by satra** [comment]
i think it's worth it as it provides a clearer notification to the user when something is missing because of an input dependency rather than the user having to parse an error either in the function or on the command line.
